urllib3/request.py

```diff
@@ -149,3 +187,24 @@ def request_encode_body(self, method, url, fields=None, headers=None,
         extra_kw.update(urlopen_kw)
 
         return self.urlopen(method, url, **extra_kw)
+
+    def redirect(self, response, method, retries, **kwargs):
+        """
+        Abstracts the redirect process to be used from any :class:`RequestMethods` object
+        """
+        url = kwargs.pop('url', '')
+        redirect_location = urljoin(url, response.get_redirect_location())
+        method = retries.redirect_method(method, response.status)
+        try:
+            pool = kwargs.pop('pool', self)
```

Comment:
Presumably this can't raise `MaxRetryError`, so let's remove it from the `try` block.